// src/app/auth/signup/signup.component.ts
import { Component, OnInit } from '@angular/core';
import { NgForm } from '@angular/forms';
// import { AuthenticationService } from '../auth.service';
import { Store } from '@ngrx/store';

import * as fromApp from '../../store/app.reducers';
import * as fromAuthActions from '../store/auth.actions';


@Component({
  selector: 'app-signup',
  templateUrl: './signup.component.html',
  styleUrls: ['./signup.component.css']
})
export class SignupComponent implements OnInit {

  constructor(private store:Store<fromApp.AppState>) { }
  // errorFlag:boolean=false;

  ngOnInit() {
  }
    onSignUp(form:NgForm){
      console.log(form);
      const email =form.value.mail;
      const password = form.value.Password;
     
      this.store.dispatch(new fromAuthActions.TrySignUp({username:email,password:password}));
      // this.authenticationService.signUp(email,password);
      
    }

    // checkValidation(form:NgForm){
    //   const emailErrors =form.value.mail;
    //   const password =form.value.Password;
    //   if(password.length<=6){
    //       this.errorFlag=true;
    //       form.controls.Password.setErrors(errors)     }
      

    // }
}
