// src/app/auth/signin/signin.component.ts
import { Component, OnInit } from '@angular/core';
import { NgForm } from '@angular/forms';
// import { AuthenticationService } from '../auth.service';
import {Store} from '@ngrx/store';
import * as fromApp from '../../store/app.reducers';
import * as fromAuthActions from '../store/auth.actions';
@Component({
  selector: 'app-signin',
  templateUrl: './signin.component.html',
  styleUrls: ['./signin.component.css']
})
export class SigninComponent implements OnInit {
  authErrorFlag:boolean=false;
  authErrormsg:string;
  constructor(
  private store:Store<fromApp.AppState>) { }

  ngOnInit() {
  }
  onSignIn(form:NgForm){
    this.authErrorFlag=false;
    this.authErrormsg='';
    console.log(form);
    const email =form.value.mail;
    const password = form.value.Password;
   
    
   this.store.dispatch(new fromAuthActions.TrySignIn({username:email,password:password}));
    // setTimeout(() =>
    //  {
    //   if(this.authenticationService.authErrorFlag){
    //     this.authErrorFlag=true;
    //     this.authErrormsg=this.authenticationService.authError;
    //   }
    // },1500)
   
    
  }

  checkErrors(){
    this.authErrorFlag=false;
    this.authErrormsg='';
  }

}
