// src/app/recipes/recipe-detail/recipe-detail.component.ts
import { Component, OnInit,Input } from '@angular/core';
import {Recipe} from '../recipe.model';
import { Store } from '@ngrx/store';
import { ActivatedRoute,Params,Router } from '@angular/router';
import { RecipieService } from '../recipie.service';
// import { Ingredient } from '../../shared/ingredient.model';
import * as shoppingListActions from '../../shopping-list/store/shopping-list.actions';
import * as fromShoppingList from '../../shopping-List/store/shopping-list.reducer';
import * as fromApp from '../../store/app.reducers';
import * as fromRecipe from '../store/recipe.reducers';
import * as fromRecipeActions from '../store/recipe.actions';
import { Observable } from 'rxjs/Observable';



// import {RecipieService } from '../recipie.service';

@Component({
  selector: 'app-recipe-detail',
  templateUrl: './recipe-detail.component.html',
  styleUrls: ['./recipe-detail.component.css']
})
export class RecipeDetailComponent implements OnInit {
 recipieState:Observable<fromRecipe.State>;
 id:number;
  constructor(private route:ActivatedRoute,
    private recipeservice:RecipieService,
  private router:Router,
private store:Store<fromRecipe.FeatureState>) { 
    
  }

  ngOnInit() {
   this.route.params.subscribe(
     (params:Params) => {
       this.id =+params['id']
      
       this.recipieState =this.store.select('recipes');

     }
   )
  }

  onEdit(){
  this.router.navigate(['edit'],{relativeTo:this.route});
    // this.router.navigate(['../',this.id, 'edit'],{relativeTo:this.route}) //alternative way
  }

  // onIngredientTransfer(){
  //   this.slService.ingredientChanged.subscribe(()
  //   {

  //   })
  // }
  onIngredientTransfer(){
    // this.recipeservice.OnAddingToShoppingList(this.recipie.ingredients)
    this.store.select('recipes')
    .take(1)
    .subscribe((recipeState:fromRecipe.State)=>{
      this.store.dispatch(new shoppingListActions.AddIngredients(recipeState.recipe[this.id].ingredients))
    })
   

  }

  onDelete(){
    this.store.dispatch(new fromRecipeActions.DeleteRecipe(this.id));
    this.router.navigate(['/recipies']);
  }
}
