// src/app/shopping-list/store/shopping-list.actions.ts
import { Action } from "@ngrx/store";
import { Ingredient } from "../../shared/ingredient.model";

export const ADD_INGREDIENT = 'ADD_INGREDIENT';
 export const ADD_INGREDIENTS ='ADD_INGREDIENTS';
 export const UPDATE_INGREDIENTS ='UPDATE_INGREDIENTS';
 export const DELETE_INGREDIENTS ='DELETE_INGREDIENTS';
 export const EDITED_INGREDIENTS ='EDITED_INGREDIENTS';
 export const STOP_EDIT ='STOP_EDIT';


export class AddIngredient implements Action {
    readonly type =ADD_INGREDIENT;
    // payload:Ingredient;
    constructor(public payload: Ingredient){}
}
export class AddIngredients implements Action {
    readonly type =ADD_INGREDIENTS;
    // payload:Ingredient;
    constructor(public payload: Ingredient[]){}
}
export class UpdateIngredients implements Action {
    readonly type =UPDATE_INGREDIENTS;
    // payload:Ingredient;
    constructor(public payload: {ingredient:Ingredient}){}
}
export class DeleteIngredients implements Action {
    readonly type =DELETE_INGREDIENTS;
    // payload:Ingredient;
   
}
export class EditedIngredients implements Action {
    readonly type =EDITED_INGREDIENTS;
    // payload:Ingredient;
    constructor(public payload: number){}
}
export class StopEdit implements Action {
    readonly type =STOP_EDIT;
    // payload:Ingredient;
    // constructor(public payload: number){}
}

export type ShoppingListActions = AddIngredient | AddIngredients | UpdateIngredients | DeleteIngredients | EditedIngredients | StopEdit;
